docs/diagrams/AddInterestSequenceDiagram.puml

```diff
@@ -0,0 +1,65 @@
+@startuml
+!include style.puml
+
+skinparam SequenceReferenceBackgroundColor White
+box Logic LOGIC_COLOR_T1
+participant ":Logic\nManager" as LogicManager LOGIC_COLOR
+participant ":AddressBook\nParser" as AddressBookParser LOGIC_COLOR
+participant ":AddInterest\nCommandParser" as AddInterestCommandParser LOGIC_COLOR
+participant ":AddInterest\nCommand" as AddInterestCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+participant "interestList:\nSet<Interest>" as Interest MODEL_COLOR
+
+end box
+
+[-> LogicManager : execute\n("addInt 1 anime")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand\n("addInt 1 anime")
+activate AddressBookParser
+
+create AddInterestCommandParser
+AddressBookParser -> AddInterestCommandParser
+activate AddInterestCommandParser
+AddInterestCommandParser --> AddressBookParser
+deactivate AddInterestCommandParser
+
+AddressBookParser -> AddInterestCommandParser : parse\n("1 anime")
+activate AddInterestCommandParser
+create Interest
+AddInterestCommandParser -> Interest
+activate Interest
+Interest --> AddInterestCommandParser
+deactivate Interest
+
+
+create AddInterestCommand
+AddInterestCommandParser -> AddInterestCommand: new\n(1, interestList)
+activate AddInterestCommand
+
+AddInterestCommand --> AddInterestCommandParser
+deactivate AddInterestCommand
+
+AddInterestCommandParser --> AddressBookParser
+deactivate AddInterestCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+AddInterestCommandParser -[hidden]-> AddressBookParser
+destroy AddInterestCommandParser
```

Comment:
I think the lifeline should not continue after the X.
UML uses an X at the __end of the lifeline__ of an object to show its deletion.
Refer to Sequence diagram --> Object Deletion: https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/uml.html#object-deletion

Reply:
PlantUML shortcomings :(
https://www.google.com/search?q=plantuml+end+lifeline+when+destroyed&oq=plantuml+end+lifeline+when+destroyed&aqs=chrome..69i57j33i160.9938j0j9&sourceid=chrome&ie=UTF-8